scripts/skills/perks/perk_rf_steady_brace.nut

```diff
@@ -0,0 +1,112 @@
+this.perk_rf_steady_brace <- ::inherit("scripts/skills/skill", {
+	m = {
+		IsSpent = true,
+		DirectDamageAddModifier = 0.2,
+		RangedSkillModifier = 20
+	},
+	function create()
+	{
+		this.m.ID = "perk.rf_steady_brace";
+		this.m.Name = ::Const.Strings.PerkName.RF_SteadyBrace;
+		this.m.Description = ::Const.Strings.PerkDescription.RF_SteadyBrace;
+		this.m.Icon = "ui/perks/perk_rf_steady_brace.png";
+		this.m.Type = ::Const.SkillType.Perk | ::Const.SkillType.StatusEffect;
+		this.m.Order = ::Const.SkillOrder.Perk;
+	}
+
+	function isHidden()
+	{
+		if (this.m.IsSpent)
+			return true;
+
+		local weapon = this.getContainer().getActor().getMainhandItem();
+		if (weapon == null)
+			return true;
+
+		foreach (s in weapon.getSkills())
+		{
+			if (this.isSkillValid(s))
+				return false;
+		}
+
+		return true;
+	}
+
+	function getTooltip()
+	{
+		local ret = this.skill.getTooltip();
+		local weapon = this.getContainer().getActor().getMainhandItem();
+		if (weapon.isWeaponType(::Const.Items.WeaponType.Crossbow))
+		{
+			ret.push({
+				id = 10,
+				type = "text",
+				icon = "ui/icons/action_points.png",
+				text = ::Reforged.Mod.Tooltips.parseString("Crossbow shots have " + ::MSU.Text.colorizeMult(1.0 + this.m.DirectDamageAddModifier, {AddSign = true}) + " armor penetration")
+			});
+		}
+		else if (weapon.isWeaponType(::Const.Items.WeaponType.Firearm))
+		{
+			ret.push({
+				id = 11,
+				type = "text",
+				icon = "ui/icons/action_points.png",
+				text = ::Reforged.Mod.Tooltips.parseString("Firearm shots have " + ::MSU.Text.colorizeValue(this.m.RangedSkillModifier, {AddSign = true, AddPercent = true}) + " chance to hit")
+			});
+		}
+
+		ret.push({
+			id = 20,
+			type = "text",
+			icon = "ui/icons/warning.png",
+			text = ::Reforged.Mod.Tooltips.parseString("Will expire upon moving or changing your weapon")
```

Comment:
`changing your weapon` is not accurate, as this effect is also removed when you swap ammunition or move items between bag and ground, in its current implementation

Furthermore `changing` is a new term to me. Previously we used the term `swapping`

I suggest to use `swapping items` here or changing the implementation of `onPayForItemAction`